File: agent_backend/model/company.go
package model

import "XWS-Project/agent_backend/dto"

type Company struct {
	Name              string                  `json:"name"`
	Email             string                  `json:"email"`
	Address           string                  `json:"address"`
	PhoneNumber       string                  `json:"phoneNumber"`
	Description       string                  `json:"description"`
	OwnerEmail        string                  `json:"ownerEmail" bson:"ownerEmail"`
	IsApproved        bool                    `json:"isApproved"`
	Comments          []string                `json:"comments"`
	InterviewReviews  []string                `json:"interviewReviews"`
	SalaryForPosition []dto.SalaryForPosition `json:"salaryForPosition"`
	Jobs              []Job                   `json:"jobs"`
}

File: agent_frontend/src/app/model/Company.ts
import { SalaryForPosition } from "./dto/SalaryForPosition";
import { Job } from "./Job";

export class Company{
    constructor(
        public name: string,
        public address: string,
        public email: string,
        public phoneNumber: string,
        public description: string,
        public isApproved: boolean,
        public ownerEmail: string,
        public comments: string[],
        public interviewReviews: string[],
        public salaryForPosition: SalaryForPosition[],
        public jobs: Job[],
    ){}
}